backend/src/main/java/com/bakdata/conquery/models/execution/ManagedExecution.java

```diff
@@ -252,36 +246,39 @@ protected void setStatusBase(@NonNull MetaStorage storage, UriBuilder url, @NonN
 	@Nullable
 	protected abstract URL getDownloadURLInternal(UriBuilder url) throws MalformedURLException, IllegalArgumentException, UriBuilderException;
 
-	public ExecutionStatus buildStatus(@NonNull MetaStorage storage, UriBuilder url, User user, DatasetRegistry datasetRegistry) {
-		return buildStatus(storage, url, user, datasetRegistry, EnumSet.noneOf(ExecutionStatus.CreationFlag.class));
-	}
-	public ExecutionStatus buildStatus(@NonNull MetaStorage storage, UriBuilder url, User user, DatasetRegistry datasetRegistry, @NonNull ExecutionStatus.CreationFlag creationFlag) {
-		return buildStatus(storage, url, user, datasetRegistry, EnumSet.of(creationFlag));
+	/**
+	 * Renders a lightweight status with meta information about this query. Computation an size should be small for this.
+	 */
+	public ExecutionStatus.Overview buildStatusOverview(@NonNull MetaStorage storage, User user, DatasetRegistry datasetRegistry) {
+		ExecutionStatus.Overview status = new ExecutionStatus.Overview();
+		setStatusBase(storage, user, status);
+
+		return status;
 	}
-	
-	public ExecutionStatus buildStatus(@NonNull MetaStorage storage, UriBuilder url, User user, DatasetRegistry datasetRegistry, @NonNull EnumSet<ExecutionStatus.CreationFlag> creationFlags) {
-		ExecutionStatus status = new ExecutionStatus();
-		setStatusBase(storage, url, user, status);
-		for(CreationFlag flag : creationFlags) {
-			switch (flag) {
-				case WITH_COLUMN_DESCIPTION:
-					setAdditionalFieldsForStatusWithColumnDescription(storage, url, user, status,  datasetRegistry);
-					break;
-				case WITH_SOURCE:
-					setAdditionalFieldsForStatusWithSource(storage, url, user, status);
-					break;
-				case WITH_GROUPS:
-					setAdditionalFieldsForStatusWithGroups(storage, user, status);
-					break;
-				default:
-					throw new IllegalArgumentException(String.format("Unhandled creation flag %s", flag));
-			}
+
+	/**
+	 * Renders an extensive status of this query (see {@link ExecutionStatus.Full}. The rendering can be computation intensive and can produce a large
+	 * object. The use  of the full status is only intended if a client requested specific information about this execution.
+	 */
+	public ExecutionStatus.Full buildStatusFull(@NonNull MetaStorage storage, UriBuilder url, User user, DatasetRegistry datasetRegistry) {
+		ExecutionStatus.Full status = new ExecutionStatus.Full();
+		setStatusBase(storage, user, status);
+
+		setAdditionalFieldsForStatusWithColumnDescription(storage, url, user, status,  datasetRegistry);
```

Comment:
setzt du jetzt immer alles anstatt wie oben mit den Flags?

Reply:
Ja ich habe abgeschätzt, dass das wohl in Ordnung ist. Ich die Flags hatte ich nur an zwei Stellen genutzt und es hatte dann komische Member erzeugt.